src/databases/functions/users/user-set-karma-score.js

```diff
@@ -0,0 +1,15 @@
+const {User} = require('../../models');
+
+module.exports = async (userId, combined_user_score, karma_score) => {
+  if (!userId) return null;
+  try {
+    const result = await User.update(
+      {combined_user_score, karma_score},
+      {where: {user_id: userId}}
+    );
+    return result;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The update operation does not check if the user exists before attempting to update. If the user does not exist, this will silently fail and return `[0]`, which might be misleading. Consider fetching the user first to ensure they exist before attempting the update.

```javascript
const user = await User.findOne({ where: { user_id: userId } });
if (!user) throw new Error(`User with id ${userId} does not exist`);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->